corehq/util/soft_assert/api.py

```diff
@@ -0,0 +1,107 @@
+from django.core.mail import mail_admins, send_mail
+from django.core.cache import cache
+from corehq.util.soft_assert.core import SoftAssert
+import settings
+
+
+def _number_is_power_of_two(x):
+    # it turns out that x & (x - 1) == 0 if and only if x is a power of two
+    # http://stackoverflow.com/a/600306/240553
+    return x > 0 and (x & (x - 1) == 0)
+
+
+def _django_caching_counter(key):
+    cache_key = 'django-soft-assert.{}'.format(key)
+    try:
+        return cache.incr(cache_key)
```

Comment:
@czue @snopoke Any reasons you can think of that this wouldn't be working in production? So far it appears to always `return 1` below though I may be misinterpreting. The symptom is all the emails I'm getting (not that many surprisingly) have "Occurrences to date: 1" at the bottom, when given that they're all the same (same key/hash thing as well), you'd expect that number to be going up.

Just raising this to you two because it would be funny if it turned out that our caching configuration on prod is broken. Maybe funny's not the right word.

(This worked as expected locally and on staging.)